nemo/collections/tts/data/vocoder_dataset.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import traceback
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import librosa
+import torch.utils.data
+
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.asr.parts.utils.manifest_utils import read_manifest
+from nemo.collections.tts.parts.preprocessing.feature_processors import FeatureProcessor
+from nemo.collections.tts.parts.utils.tts_dataset_utils import (
+    filter_dataset_by_duration,
+    get_abs_rel_paths,
+    get_weighted_sampler,
+    stack_tensors,
+)
+from nemo.core.classes import Dataset
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+
+@dataclass
+class DatasetMeta:
+    manifest_path: Path
+    audio_dir: Path
+    sample_weight: float = 1.0
+
+
+@dataclass
+class DatasetSample:
+    manifest_entry: dict
+    audio_dir: Path
+
+
+@experimental
+class VocoderDataset(Dataset):
+    def __init__(
+        self,
+        dataset_meta: Dict[str, DatasetMeta],
+        sample_rate: int,
+        n_segments: Optional[int] = None,
+        weighted_sample_steps: Optional[int] = None,
+        feature_processors: Optional[Dict[str, FeatureProcessor]] = None,
+        min_duration: Optional[float] = None,
+        max_duration: Optional[float] = None,
+        num_audio_retries: int = 5,
+    ):
+        super().__init__()
+
+        self.sample_rate = sample_rate
+        self.n_segments = n_segments
+        self.weighted_sample_steps = weighted_sample_steps
+        self.num_audio_retries = num_audio_retries
+        self.load_precomputed_mel = False
+
+        if feature_processors:
+            logging.info(f"Found feature processors {feature_processors.keys()}")
+            self.feature_processors = feature_processors.values()
```

Comment:
(A nitpick)

`.values()` [doesn't return a list](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects)

If anyone ever does something like this:
```
feature_processors = ...
dataset1 = VocoderDataset(..., feature_processors, ...)
feature_processors["another_processor"] = ...
dataset2 = VocoderDataset(..., feature_processors, ...)
```
then, both datasets would keep reference to the original collection, and would have the same list of feature processors. Which is likely not intended.

Turning it into a list will make it a bit more foolproof:
```
self.feature_processors = list(feature_processors.values())
```